demos/http/http_demo_s3_upload/http_demo_s3_upload.c

```diff
@@ -0,0 +1,624 @@
+        {
+            LogError( ( "Failed to upload the data to S3. The file size found is %d, but it should be %d.",
+                        ( int32_t ) *pFileSize,
+                        ( int32_t ) DEMO_HTTP_UPLOAD_DATA_LENGTH ) );
+        }
+        else
+        {
+            LogInfo( ( "Successfuly verified that the size of the file found on S3 matches the file size uploaded "
+                       "(Uploaded: %d bytes, Found: %d bytes).",
+                       ( int32_t ) DEMO_HTTP_UPLOAD_DATA_LENGTH,
+                       ( int32_t ) *pFileSize ) );
+        }
+    }
+    else
+    {
+        LogError( ( "An error occurred in getting the file size from %s. Error=%s.",
+                    serverHost,
+                    HTTPClient_strerror( httpStatus ) ) );
+    }
+
+    return httpStatus;
+}
+
+static HTTPStatus_t uploadS3ObjectFile( const TransportInterface_t * pTransportInterface,
+                                        const char * pPath )
+{
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+
+    /* Initialize the request object. */
+    requestInfo.pHost = serverHost;
+    requestInfo.hostLen = serverHostLength;
+    requestInfo.method = HTTP_METHOD_PUT;
+    requestInfo.methodLen = HTTP_METHOD_PUT_LENGTH;
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = strlen( pPath );
+
+    /* Set "Connection" HTTP header to "keep-alive" so that multiple requests
+     * can be sent over the same established TCP connection. */
+    requestInfo.reqFlags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
+
+    /* Set the buffer used for storing request headers. */
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    /* Initialize the response object. The same buffer used for storing
+     * request headers is reused here. */
+    response.pBuffer = userBuffer;
+    response.bufferLen = USER_BUFFER_LENGTH;
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders,
+                                                          &requestInfo );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        LogInfo( ( "Uploading file..." ) );
+        LogInfo( ( "Request Headers:\n%.*s",
+                   ( int32_t ) requestHeaders.headersLen,
+                   ( char * ) requestHeaders.pBuffer ) );
+        httpStatus = HTTPClient_Send( pTransportInterface,
+                                      &requestHeaders,
+                                      ( const uint8_t * ) DEMO_HTTP_UPLOAD_DATA,
+                                      DEMO_HTTP_UPLOAD_DATA_LENGTH,
+                                      &response,
+                                      0 );
+    }
+    else
+    {
+        LogError( ( "Failed to initialize HTTP request headers: Error=%s.",
+                    HTTPClient_strerror( httpStatus ) ) );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        LogInfo( ( "Received HTTP response from %s%s...",
```

Comment:
Can we please change these printings of the response headers and body to LogDebug and then have a LogInfo() about the success from a 200 response?